lib/ansible/plugins/inventory/docker_machine.py

```diff
@@ -0,0 +1,227 @@
+# (https://docs.docker.com/machine/drivers/), e.g.:
+#   $ docker-machine create --driver digitalocean ... mymachine
+#   $ ansible-inventory -i ./path/to/docker-machine.yml --host=mymachine
+#   {
+#     ...
+#     "digitalocean": {
+#       "hosts": [
+#           "mymachine"
+#       ]
+#     ...
+#   }
+strict: no
+keyed_groups:
+  - separator: ''
+    key: docker_machine_node_attributes.DriverName
+
+# Example grouping hosts by Digital Machine tag
+strict: no
+keyed_groups:
+  - prefix: tag
+    key: 'dm_tags'
+
+# Example using compose to override the default SSH behaviour of asking the user to accept the remote host key
+compose:
+  ansible_ssh_common_args: '"-o StrictHostKeyChecking=accept-new"'
+'''
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible.module_utils._text import to_text
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+from ansible.utils.display import Display
+
+import json
+import re
+import subprocess
+
+display = Display()
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    ''' Host inventory parser for ansible using Docker machine as source. '''
+
+    NAME = 'docker_machine'
+
+    def _run_command(self, args):
+        command = ['docker-machine']
+        command.extend(args)
+        display.debug('Executing command {0}'.format(command))
+        try:
+            result = subprocess.check_output(command)
+        except Exception as e:
+            display.warning('Exception {0} caught while executing command {1}, this was the original exception: {2}'.format(type(e).__name__, command, e))
+            raise e
+
+        return result.decode('utf-8').strip()
+
+    def _get_docker_daemon_variables(self, id):
+        '''
+        Capture settings from Docker Machine that would be needed to connect to the remote Docker daemon installed on
+        the Docker Machine remote host. Note: passing '--shell=sh' is a workaround for 'Error: Unknown shell'.
+        '''
+        try:
+            env_lines = self._run_command(['env', '--shell=sh', id]).splitlines()
+        except Exception:
+            # This can happen when the machine is created but provisioning is incomplete
+            return None
+
+        # example output of docker-machine env --shell=sh:
+        #   export DOCKER_TLS_VERIFY="1"
+        #   export DOCKER_HOST="tcp://134.209.204.160:2376"
+        #   export DOCKER_CERT_PATH="/root/.docker/machine/machines/routinator"
+        #   export DOCKER_MACHINE_NAME="routinator"
+        #   # Run this command to configure your shell:
+        #   # eval $(docker-machine env --shell=bash routinator)
+
+        # capture any of the DOCKER_xxx variables that were output and create Ansible host vars
+        # with the same name and value but with a dm_ name prefix.
+        vars = []
+        for line in env_lines:
+            match = re.search('(DOCKER_[^=]+)="([^"]+)"', line)
```

Comment:
assumes native strings in line, you should convert to_native or ensure regex matches 'line' string type

Reply:
When to_text is used in self._run_command() the lines will be text, so this will be okay because line and the regex will both be text, correct?

Reply:
I guess you can add `from __future__ import unicode_literals`, then this should be no problem. @bcoca is that correct?

Reply:
I've resolved the other conversations (assuming that @bcoca is okay with the changes), so that just leaves this point for which I am grateful for your assistance as I'm not sure if this is resolved by the changes i just pushed or if as @felixfontein suggests that import and/or other tweaks are still needed.

Reply:
I am resolving this point as @felixfontein stated that he is okay to consider this ready, subject to Python 2 testing which I just successfully performed.

Reply:
Resolving as @felixfontein is happy with the current state of the PR. @bcoca please of course feel free to come back to me with more improvements/issues/etc.